dlio_benchmark/utils/utility.py

```diff
@@ -51,12 +56,52 @@ def utcnow(format=LOG_TS_FORMAT):
     return datetime.now().strftime(format)
 
 
-def get_rank():
-    return MPI.COMM_WORLD.rank
-
-
-def get_size():
-    return MPI.COMM_WORLD.size
+class mpi:
+    __instance = None
+
+    def __init__(self):
+        if mpi.__instance is not None:
+            raise Exception(f"Class {self.__init__.__qualname__} is a singleton!")
+        else:
+            self.mpi_rank = -1
+            self.mpi_size = -1
```

Comment:
Initialize this to 1 by default.

Reply:
These two cannot change, they are necessary for detecting a process which tries to use its rank or comm_size before DLIOMPI.initialize() has been called (in the benchmark parent process case) or before the parent's rank and comm_size have been initialized by DLIOMPI.set_parent_values() in a pytorch read_thread child process.  See the checks in rank(), size(), and comm().  read_threads children created via 'fork' are especially tricky, they appear to have been MPI-initialized because they are a fork, but they need to be prevented from making calls like barrier() because they are not members of the MPI job, they don't have their own rank values, and comm_size doesn't include them.   They can however use their parent's rank and comm_size values for math related to deciding which samples to read or whatever.

Reply:
I see. But wouldn't having an explicit variable to detect this would be better. Repurposing rank and comm size for this could lead to create bugs in the future. 

I would recommend adding a variable which can explicitly track this.

Reply:
Yes, I was thinking about this as well.  Having child processes using the same rank() and comm() functions is confusing.  It would be better if they were clearly retrieving their parent's values.  I'll work on this.